endpoint/endpoint.go

```diff
@@ -217,6 +226,13 @@ func New(config Config) (*Endpoint, error) {
 	return e, nil
 }
 
+//RegisterResponse type gives access to metric id and name
```

Comment:
Why duplicate data_registry.RegisterMetricsResponse ?

Reply:
RegisterMetricsResponse comes down eventually to an array of RegisterMetricResponse or error. This in turn is instanceId, definitionId, and name. Maybe an instanceId is the equivalent of metric id? Or is that definitionId? I felt that it was more clear to directly just have a metric id.

Reply:
InstanceId is the metricId. I can see how using metricId is less ambiguous, but the reason for reusing the data-registry constructs is to provide, as far as possible,  the same interface as seen if writing the the ZC API directly. xdm is just a proxy for that.